src/vm/types/mod.rs

```diff
@@ -726,47 +726,56 @@ impl Value {
         if let Value::UInt(inner) = self {
             inner
         } else {
-            panic!(format!("Value '{:?}' is not a u128", &self));
+            error!("Value '{:?}' is not a u128", &self);
+            panic!();
         }
     }
 
     pub fn expect_i128(self) -> i128 {
         if let Value::Int(inner) = self {
             inner
         } else {
-            panic!(format!("Value '{:?}' is not an i128", &self));
+            error!("Value '{:?}' is not an i128", &self);
+            panic!();
         }
     }
 
     pub fn expect_buff(self, sz: usize) -> Vec<u8> {
-        if let Value::Sequence(SequenceData::Buffer(buffdata)) = self {
-            if buffdata.data.len() == sz {
+        if let Value::Sequence(SequenceData::Buffer(mut buffdata)) = self {
+            if buffdata.data.len() <= sz {
+                for _ in buffdata.data.len()..sz {
+                    buffdata.data.push(0u8)
```

Comment:
Since these `expect_` functions are basically destructuring routines, it makes more sense for data padding to happen in the caller rather than in this function (a destructuring routine shouldn't mutate the returned data).

Reply:
Fair enough.  Will fix.